controllers/authrealm/dex.go

```diff
@@ -399,35 +400,65 @@ func (r *AuthRealmReconciler) createDexConnectors(authRealm *identitatemv1alpha1
 			r.Log.Info("generated intermediate connectors", "cs", cs)
 		case openshiftconfigv1.IdentityProviderTypeLDAP:
 			r.Log.Info("create connector for LDAP")
+			url, err := ldaputil.ParseURL(idp.LDAP.URL)
+			if err != nil {
+				return nil, fmt.Errorf("Error parsing LDAP URL: %v", err)
+			}
+			var scope string
+			switch url.Scope {
+			case ldaputil.ScopeSingleLevel:
+				scope = "one"
+			case ldaputil.ScopeWholeSubtree:
+				scope = "sub"
+			}
+
 			c := &dexoperatorv1alpha1.ConnectorSpec{
 				Type: dexoperatorv1alpha1.ConnectorTypeLDAP,
 				Name: idp.Name,
 				Id:   idp.Name,
 				LDAP: dexoperatorv1alpha1.LDAPConfigSpec{
-					Host:   idp.LDAP.URL,
+					Host:   url.Host,
 					BindDN: idp.LDAP.BindDN,
 					BindPWRef: corev1.SecretReference{
 						Name:      idp.LDAP.BindPassword.Name,
 						Namespace: authRealm.Namespace,
 					},
 					InsecureNoSSL:      false,
-					InsecureSkipVerify: idp.LDAP.Insecure,
+					InsecureSkipVerify: false,
 					UsernamePrompt:     "Email Address",
 					RootCARef: corev1.SecretReference{
 						Name:      idp.LDAP.CA.Name,
 						Namespace: authRealm.Namespace,
 					},
 					UserSearch: dexoperatorv1alpha1.UserSearchSpec{
-						BaseDN:    authRealm.Spec.LDAPExtraConfigs[idp.Name].BaseDN,
-						Filter:    authRealm.Spec.LDAPExtraConfigs[idp.Name].Filter,
-						Username:  idp.LDAP.Attributes.PreferredUsername[0],
+						BaseDN:    url.BaseDN,
+						Filter:    url.Filter,
+						Scope:     scope,
+						Username:  url.QueryAttribute,
 						IDAttr:    idp.LDAP.Attributes.ID[0],
 						EmailAttr: idp.LDAP.Attributes.Email[0],
 						NameAttr:  idp.LDAP.Attributes.Name[0],
 					},
 					GroupSearch: authRealm.Spec.LDAPExtraConfigs[idp.Name].GroupSearch,
 				},
 			}
+			switch url.Scheme {
+			case "ldap":
+				if idp.LDAP.Insecure {
```

Comment:
This configuration of insecure connection, not upgraded leaks passwords, and per Dex docs may not be supported in the future. Shall we perhaps not support this option? We could update our webhook to never allow `insecure: true` with our operator.

Reply:
Yeah I agree with not allowing `insecure: true`. I will update the webhook check to not allow `insecure: true`